### PR TITLE
Fix bug with create edit

### PR DESCRIPTION
**Pull request by NoaBablikiCertora**
fixed the bug with create / edit according to our conversations.
the titles on the `save edit` might look a little confusing (didn't change them, they looked confusing before...) but we are going to change the entire view anyway... so I think its ok...

### CODE REVIEW

**Comment by NoaBablikiCertora** [comment]
done removing the file name from the save button title.
ready for review :)

File: src/extension/SettingsPanel.ts [hunk 1]
```diff
@@ -4,12 +4,15 @@ import { getNonce } from './utils/getNonce'
 import { createAndOpenConfFile } from './utils/createAndOpenConfFile'
 import { log, Sources } from './utils/log'
 import { CommandFromSettingsWebview, EventFromSettingsWebview } from './types'
+import { settings } from 'cluster'
```
Comment (fnast9) [inline src/extension/SettingsPanel.ts:7]: What's this? Where do you use it?

File: src/extension/SettingsPanel.ts [hunk 2]
```diff
@@ -60,33 +64,75 @@ export class SettingsPanel {
...
+      editConfFile,
+    )
+    this.allPanels.push(SettingsPanel.currentPanel)
+  }
+
+  /**
+   * renders a settings panel:
+   * if the settings panel is not an edit: create a new panel in a new tab
+   * else if the settings panel for that edit was not created, or created and deleted: create new
+   * else: open the already created panel tab
```
Comment (fnast9) [inline src/extension/SettingsPanel.ts:106]: It's great that you documented it. Let's make it a bit shorter:
Open a new webview panel when creating or editing conf files. If the panel is already opened (for editing an existing configuration file), show the same panel.

File: src/extension/SettingsPanel.ts [hunk 3]
```diff
@@ -60,33 +64,75 @@ export class SettingsPanel {
...
+   * @param extensionUri uri of the extension folder
+   * @param editConfFile conf file to edit
```
Comment (fnast9) [inline src/extension/SettingsPanel.ts:108]: Please replace with `conf file content`

**Review by fnast9 — CHANGES_REQUESTED** [review]
Great job! :+1: 
Please see the comments

File: src/extension/SettingsPanel.ts [hunk 4]
```diff
@@ -60,33 +64,75 @@ export class SettingsPanel {
...
+      })
+    }
+    if (isOpened && SettingsPanel.currentPanel) {
+      try {
+        SettingsPanel.currentPanel._panel.reveal(vscode.ViewColumn.One)
+      } catch (e) {
+        // delete closed panel from the array and render again
+        SettingsPanel.allPanels = SettingsPanel.allPanels.filter(
+          p => p !== SettingsPanel.currentPanel,
+        )
```
[the remark this answers is not in the document]
Comment (NoaBablikiCertora) [reply src/extension/SettingsPanel.ts:131]: I tried but when I did, the "this" instance was "null" so I could not reach the closed tab...

File: src/extension/SettingsPanel.ts (hunk 1 above)
Comment (NoaBablikiCertora) [reply src/extension/SettingsPanel.ts:7]: nowhere this is a mistake I'll remove it

File: src/extension/SettingsPanel.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (fnast9) [reply src/extension/SettingsPanel.ts:129]: Let's give it another try. Look for `this._panel.onDidDispose` function again, and try to replace `this.dispose` with an anonymous function. Use the `console.log` to check `this` (object) and see if you can locate it in the `allPanels`.

**Review by fnast9 — CHANGES_REQUESTED** [review]
Thanks for addressing my comments. There is only one left.

File: src/extension/SettingsPanel.ts [hunk 5]
```diff
@@ -63,7 +63,16 @@ export class SettingsPanel {
...
+    this._panel.onDidDispose(
+      this.removeFromAllPanelsAndDispose,
+      null,
+      this._disposables,
+    )
+  }
+
+  removeFromAllPanelsAndDispose = (): void => {
+    SettingsPanel.allPanels = SettingsPanel.allPanels.filter(p => p !== this)
+    this.dispose()
```
Comment (fnast9) [inline src/extension/SettingsPanel.ts:75]: It would be nice to have a test to check the panels' list.

**Review by fnast9 — APPROVED** [review]
Thank you, Noa!
Please, add this PR to the CHANGELOG.md and update the release date.
Also, please merge the main branch into yours before you merge this PR.

**Pull request merged**